### ISSUE TITLE
Allow to specify toctree pages as direct-children of parent page

### ISSUE THREAD

**Issue by choldgraf** [sphinx-doc/sphinx#8287]
Currently, all pages listed in a `toctree` directive will be listed as children of the last header level under which the `toctree` is placed. For example:

```rst
My title
========

My header
---------

.. toctree::

   page1
```

Will turn into:

```
1. My title
1.1 My header
1.1.1 page1 title
```

However, I find that many users use the `index` page similar to an "introduction" to the chapter, and put headers on that page. They really wish for the toctree to be treated as *siblings* of the index page top-level headers, rather than as children of the latest header. E.g. they want the above example to render instead as:

```
1. My title
1.1 My header
1.2 page1 title
```

I can get close to what they want with the following:

```rst
My title
========

.. toctree::

   page1

My header
---------

```

however, this results in an incorrect ordering of the sections (which becomes a problem with latex builds etc).

I wonder if this pattern is possible right now and I am missing something? Otherwise, I'm curious what folks think about adding a keyword argument for `toctree` like `level: 2` that would tell Sphinx what level it should use for the toctree titles.

I'm not sure if I'm explaining this well or not, happy to try and clarify if it helps!

**Comment by cgay** [comment]
In Sphinx 4.2.0 I don't see the behavior described at the top of the first comment at all. No matter where I put the `toctree` directive in my top-level `index.rst` page, none of the headers in `index.rst` appear in the ToC at all; only the headers in the files listed in the `toctree` directive's contents show up in the ToC.

For example, given this `index.rst` file, 

```
$ cat index.rst 
.. abc documentation master file, created by
   sphinx-quickstart on Sun Oct 10 23:06:59 2021.
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

Welcome to abc's documentation!
===============================

Dog
===

My dog has fleas.

.. toctree::

   foo
   bar

Cat
---

My cat has paws.

CatSub
~~~~~~

This is a cat subheader.

CatSubSub
.........

This is a cat subsubheader.


Indices and tables
==================

* :ref:`genindex`
* :ref:`modindex`
* :ref:`search`
```

The sidebar looks like this:

```
abc                  // project name
Navigation           // boilerplate?
1. Foo               // Title in foo.rst
2. Bar               // Title in bar.rst
QuickSearch
_________
```

Specifically, **none** of the headers in `index.rst` are in the sidebar, no matter where I place the `toctree` directive. 

Sphinx 4.2.0, Alabaster theme. Hopefully this isn't just a n00b mistake on my part. My test site is attached.
[site.tar.gz](https://github.com/sphinx-doc/sphinx/files/7319026/site.tar.gz)

**Comment by jwodder** [comment]
Although the original post suggests HTML, I am experiencing this phenomenon when building a PDF of my docs via LaTeX using Sphinx 5.0.1.  The pages listed in the toctree are treated as sub-sections of whichever section in `index.rst` the directive occurs under rather than as siblings, and I cannot see a way to get what I want.

**Comment by ES-Alexander** [comment]
I've just run into this issue (albeit not in a top level index page), and was surprised it seems to still not be possible.

If I'm understanding correctly, it seems like #8518 discusses the same problem.

**Comment by Paebbels** [comment]
I also see it in PDF files. It would be nice to reset the current nesting level with a directive like

```rst
.. toc-level-up:: 2
```

The current way how `toctree` works, limits how to decompose bigger documentations into multiple files and pages. Sometimes one needs to add hundreds of lines into one file, just to have a well-structured navigation bar, but the ReST file becomes hard to maintain.

**Comment by jdillard** [comment]
I agree that authors should have more flexibility in controlling where a `toctree` is attached within the TOC hierarchy.

For a local option, if the value is absolute, I think `:target-level:` would make that behavior clearer than `:level:`.

However, I would prefer a relative `:level-up:` value specifying how many section levels to ascend. This seems easier for authors because it relies only on the local section context rather than requiring knowledge of the full TOC hierarchy.

```rst
My title
========

My header
---------

.. toctree::
   :level-up: 1

   page1
```

Currently, the referenced document is nested beneath `My header`:

```text
My title
└── My header
    └── Page 1 title
```

With `:level-up: 1`, the two entries would instead be siblings:

```text
My title
├── My header
└── Page 1 title
```

The proposed stateful `toc-level-up` directive would be a desirable addition for documents containing several `toctree` directives. It could follow the pattern of [`highlight`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-highlight), affecting subsequent `toctree` directives in the same document.

A stateful form and a local option could coexist, as `:level-up:` on an individual `toctree` would override `toc-level-up`.

Some semantics would need to be specified before implementation, for example:

* The level change should be measured from the section containing each `toctree`, not from the section containing the `toc-level-up` directive.
* Promotion beyond the available ancestors should produce a warning rather than silently clamp the level.
* The resulting hierarchy and ordering should be identical for HTML, singlehtml, LaTeX/PDF, and other builders, including hidden toctrees.

**Comment by jdillard** [comment]
I turned this into an extension if people want to try it out: https://github.com/jdillard/sphinx-toctree-level-up

I figured that might make it easier to use the feature and ultimately get it upstreamed.
